petab/lint.py

```diff
@@ -622,7 +623,7 @@ def observable_table_has_nontrivial_noise_formula(
     """
 
     return not observable_df[NOISE_FORMULA].apply(
-        lambda x: isinstance(x, numbers.Number) or
+        lambda x: is_scalar_float(x) or
         re.match(r'^[\w_\.]+$', str(x)) is not None
```

Comment:
```suggestion
        re.match(r'^\w+$', str(x)) is not None
```
If numbers are handled separately, we can remove the dot. Underscore should be in \w already.

Reply:
👍